realm-annotations-processor/src/main/java/io/realm/processor/ModuleMetaData.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.processor;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+
+import io.realm.annotations.internal.RealmModule;
+
+/**
+ * Utility class for holding metadata for the Realm modules.
+ */
+public class ModuleMetaData {
+
+    private final Set<ClassMetaData> availableClasses;
+    private final RoundEnvironment env;
+    private Map<String, Set<ClassMetaData>> modules = new HashMap<String, Set<ClassMetaData>>();
+    private Map<String, Set<ClassMetaData>> libraryModules = new HashMap<String, Set<ClassMetaData>>();
+    private Map<String, ClassMetaData> classMetaData = new HashMap<String, ClassMetaData>();
+    private boolean shouldCreateDefaultModule;
+
+    public ModuleMetaData(RoundEnvironment env, Set<ClassMetaData> availableClasses) {
+        this.env = env;
+        this.availableClasses = availableClasses;
+        for (ClassMetaData metadata : availableClasses) {
+            classMetaData.put(metadata.getFullyQualifiedClassName(), metadata);
+        }
+    }
+
+    public boolean generate(ProcessingEnvironment processingEnv) {
+
+        // Check that modules are setup correctly
+        for (Element classElement : env.getElementsAnnotatedWith(RealmModule.class)) {
+            if (!classElement.getKind().equals(ElementKind.CLASS)) {
+                Utils.error("The RealmModule annotation can only be applied to classes", classElement);
+                return false;
+            }
+
+            RealmModule module = classElement.getAnnotation(RealmModule.class);
+            Utils.note("Processing " + classElement.getSimpleName());
```

Comment:
`classElement.getSimpleName()` is used in several instances. Better extract it to a local variable